app/server.rb

```diff
@@ -0,0 +1,103 @@
+require 'sinatra/base'
+require 'sinatra/partial'
+require './app/models/user'
+require './app/models/message'
+require 'data_mapper'
+require 'dm-validations'
+require './app/data_mapper_setup'
+require 'rack-flash'
+  
+
+
+  class Chitter < Sinatra::Base
+    enable :sessions
+    use Rack::Flash
+    use Rack::MethodOverride
+    register Sinatra::Partial
+    set :partial_template_engine, :erb
+   
+
+
+  get '/' do
+    @messages = Message.all
+    erb :index
+  end
+    
+  get '/users/new' do
+    erb :'users/new'
+  end
+
+  get '/sessions/new' do
+    erb :'sessions/new'
+  end
+
+  post '/users/new' do
```

Comment:
often with a large server file like this we extract these routes into controller files to keep things a bit tidier here